server/src/main/java/org/opensearch/indices/SystemIndexRegistry.java

```diff
@@ -56,14 +56,25 @@ static void register(Map<String, Collection<SystemIndexDescriptor>> pluginAndMod
             .collect(Collectors.toList());
         descriptors.add(TASK_INDEX_DESCRIPTOR);
 
-        SYSTEM_INDEX_DESCRIPTORS = descriptors.stream().collect(Collectors.toUnmodifiableList());
+        SYSTEM_INDEX_DESCRIPTORS_MAP = descriptorsMap;
         SYSTEM_INDEX_PATTERNS = descriptors.stream().map(SystemIndexDescriptor::getIndexPattern).toArray(String[]::new);
```

Comment:
Do we even need `descriptors`? Could we use `descriptorsMap` ?

Reply:
Removed `descriptors`. That data structure was also duplicated. The constructor is a lot simpler now.